modules/systemd.nix

```diff
@@ -73,22 +75,87 @@ let
 
   servicesStartTimeoutMs = builtins.toString cfg.servicesStartTimeoutMs;
 
-  unitType =
-    unitKind:
-    with types;
-    let
-      primitive = oneOf [
-        bool
-        int
-        str
-        path
+  unitBaseType =
+    unitKind: mod:
+    types.submodule {
+      freeformType =
+        with types;
+        let
+          primitive = oneOf [
+            bool
+            int
+            str
+            path
+          ];
+        in
+        attrsOf (attrsOf (either primitive (listOf primitive)))
+        // {
+          description = "systemd ${unitKind} unit configuration";
+        };
+
+      imports = [
+        {
+          options.Unit = {
+            Description = mkOption {
+              type = types.nullOr types.str;
+              default = null;
+              example = "My daily database backup";
+              description = "A short human-readable label of the unit.";
+            };
+
+            Documentation = mkOption {
+              type = with types; either str (listOf str);
+              apply = p: if builtins.isList p then p else [ p ];
+              example = [ "my-${unitKind}.${unitKind}" ];
+              default = [ ];
+              description = "List of URIs referencing documentation for the unit.";
+            };
+
+            # After = mkOption {
+            #   type = types.listOf types.str;
+            #   example = [ "my-${unitKind}.${unitKind}" ];
+            #   default = [ ];
+            #   description = "Units that should be started before this one.";
+            # };
+          };
+        }
+
+        mod
       ];
-    in
-    attrsOf (attrsOf (attrsOf (either primitive (listOf primitive))))
-    // {
-      description = "systemd ${unitKind} unit configuration";
     };
 
+  unitType = unitKind: types.attrsOf (unitBaseType unitKind { });
+
+  serviceType = types.attrsOf (
+    unitBaseType "service" {
+      options.Service = {
+        Environment = mkOption {
+          type = with types; either str (listOf str);
```

Comment:
It would probably be possible to support something like `attrsOf primitive` as a more ergonomic way to express variable assignment (similar to the `sessionVariables` options).